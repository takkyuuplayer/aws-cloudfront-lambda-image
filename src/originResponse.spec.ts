import fs from "fs";
import sharp from "sharp";
import * as originResponse from "./originResponse";

describe("originResponse", () => {
    const svg = fs.readFileSync(__dirname + "/../test/data/android.svg", "utf8");

    describe("convert", () => {
        it("returns original image if params is not present", async () => {
            const buf = Buffer.from(svg);
            return originResponse.convert(buf, {})
                .then((data) => data.toString("utf-8", 0, data.length))
                .then((data) => expect(data).toStrictEqual(svg));
        });

        it("can change format", async () => {
            const buf = Buffer.from(svg);
            const png = fs.readFileSync(__dirname + "/../test/data/android.png");
            return originResponse.convert(buf, { format: "png" })
                .then((data) => expect(data).toStrictEqual(png));
        });

        it("can change size", async () => {
            const buf = Buffer.from(svg);
            const png = fs.readFileSync(__dirname + "/../test/data/android.300x400.png");
            return originResponse.convert(buf, { format: "png", size: "300x400" })
                .then((data) => expect(data).toStrictEqual(png));
        });

        it("can change color", async () => {
            const buf = Buffer.from(svg);
            const png = fs.readFileSync(__dirname + "/../test/data/android.greyscale.png");
            return originResponse.convert(buf, { format: "png", color: "greyscale" })
                .then((data) => expect(data).toStrictEqual(png));
        });

        it("returns png if the format is not supported", async () => {
            const buf = Buffer.from(svg);
            const png = fs.readFileSync(__dirname + "/../test/data/android.png");
            return originResponse.convert(buf, { format: "gif" })
                .then((data) => expect(data).toStrictEqual(png));
        });
    });
});
