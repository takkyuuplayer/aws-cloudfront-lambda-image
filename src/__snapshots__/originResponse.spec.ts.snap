// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`originResponse handle convert png to jpeg 1`] = `
Object {
  "body": "/9j/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCABkAGQDASIAAhEBAxEB/8QAHAABAAIDAQEBAAAAAAAAAAAAAAUGAQcIBAID/8QARRAAAQMDAQMGCQcKBwAAAAAAAQACAwQFEQYHEiETMUFRcbEIFjI1YXSRk+EUIlVzgbLBFSMzNDdCUnWhsyQ2ZYKDotH/xAAaAQABBQEAAAAAAAAAAAAAAAAAAQIDBQYE/8QALxEAAgEDAgMGBAcAAAAAAAAAAAECAwQRBTEhcbESMzRBUWEUoeHwBhMiMoGRwf/aAAwDAQACEQMRAD8A5UREQAW7fB42bXy53qh1g+sNjstunErK17cuqHNPFkbT5QPFpJ4cSOJyFXdg+zxuvtWltxLo7BbmCpuEoOMt/djB6C4g9gDjz4XVN0r21XI09JE2mttM0RU1NG3dZGwDA4D0exV+oX8bOGd5PZCN4JSq1I+nuFbJZhyUNSQ48o0HDhzuA6M+lRNTdK+pJM9ZUP8ARvkD2Be12nqlthF0MkfJlofyfTu558832KGWRuatysKq2k+KXlx9iJtn6tqahhy2omaesSH/ANUnQ6lutIRipMzP4ZhvD286h16rXQy3KvipIC0PkJ4u5gAMkqKjVrRklSk8v0BNnm1vYn6/2e1WmrbdjbLrPOKiQzNzHWuBJEbnDiBndPDj80cCuLdT2C56YvdTab5SSUlfTu3Xxv8A6EHmIPOCOBXcF7tc1orhTzPa8loe17OGR+HMq9tP0dFtR0fLCGMGq7XE6Whnxh1QwcTC49OejqOD0uzpdM1BqXwldYkuC+vv1HxfkzitF9PY6N7mPaWuacEEYIPUvlXw8IiIALK2lsp2M3bXdA+71NZBZtPscWfLahpcZXA4IjZkb2OYkkDPDiQQr1evBrZNRSO0hqunuVexpcKSpp+QMuOhrt4jPaMdZCjdWEZKDksvyAvWxqzs01sUtQa0NrL9K6und0mMHDB2boaftKnlY6PTM7dKaWo5JqWikorVBA+Gofuua8MAdw7RhY8VpPpO2+9+CyuqW9evcyko8FwRHJNshjXVRoxSGol+TA5EW981edWHxWk+k7b734J4rSfSdt978FwOyuZbx+f1G4ZXl9wyyQStlhe6ORhy1zTggqe8VpPpO2+9+CeK0n0nbfe/BIrG4Tyo/NBhkHVVM1XO6aplfLKedzjkr7t9W+hroKqM4dE4O7R0j2ZUz4rSfSdt978EOlpMec7b734JytLpS7eOO+6Fwzk3wmdORae2sXF1IwMpLmxlxiAGB+czv/8AcPP2rVS7I25bJrtr7UmmnUlRS0lBR20x1lymOY24f80NA4uJyT0DrI4KlVvg1UE9MY7BraCouQHzYqmkMUch6g4OJHsK2/50EoqbSb8skuTmxFK6nsFy0xfKu0XulfS19M7dkjd7QQeYgjBBHAgopQO47pRxWe1WKy0YDKKhoImRsHAE4wXH0nH9So6OR8MjZYnFsjDvNcOcEKb1d+vUnqkXcVBO8k9iweoybupvPmRS3LNrx/LVtvkcBl9K1x+0kqs7o6grJrX9Na/U2d6ribqHG5m398BJbmN0dQTdHUFlFxCGN0dQTdHUFlEAY3R1BN0dQWUQBZ9UVDxYbHTNOInQB7gOkgDHeVWFYtU+bLD6r+DVXV237brvPouiFe5Mah2f2DXH5Nu1/oo6it+RshMhHEhrnHj7Sittk8xW76kd5Rbe2blRg36LoSrYq2rv16k9Ui7ioJ3knsU7q79epPVIu4qCd5J7FiNR8TU5kctyya1/TWv1Nneq4rHrX9Na/U2d6riNQ8RL+OiEe4REXGIEREAEREAWLVPmyw+q/g1V1WLVPmyw+q/g1V1dd9375LohXubFsnmK3fUjvKJZPMVu+pHeUW3te4hyXQlWxR3XGLUumLDqS3kSUdXRRteW8eSkaMOYeog5HaF5qSmlrahlPTML5ZDgAd/YuUNm21LUuz50sdlqIprfM7eloatnKQvPNvYyC046QRnhnOFdNQeEhquvt0tJZ6G1WMytLX1FFEeV4/wucSG9uMjoKqrnRFXruqpYT3Q1xy8nT2vo+Rr6CLOdylDc9hIVYXqFR8u0ho6tbI6Vk1np/wA4SSXODBvZJ6cnivKqDU12bqa+9hstwiIuEaEREAERYPAIAtupaKWXTlmq4mlzIYA2TA8kEDB7OCqkbXSyNjiaXyOOGtaMklUfb9tCv2z/AFRpJ2n60RTi1f4qklG/FK0v+bvs/wBrsEYI48eKo9x8JjVEtG+O12ixWyreMOqoIHOePS0OJA+0Famto3xEo1VLGUs8PYk7OTd2udrFj2d3Ci0/dHPlrYqOOSURN3uTLi47pxzHGD2EdaLiS5V9Xc6+orrhUS1NXUPMks0ri5z3HnJJRX0IKEVFbIeeVERPA3rsW2y0en7I3S+tKeoqbIx5fSVUHGWkJOS3H7zMknhxGTwIOBvPSd60RrCufRaa1NJV1jYHVHIGkexwY3GSS5oHSFwut4eCD+1Gt/lFR96NcdzZUK/66kcsRpM3qOICkbDb2XO4imklMTdxzy8DOMKOb5I7FO6M89/8EncsTZwjUrwhLZtES3InT1w0lqinubtL6iNxmoIuVlYKZ8eAc44uA5yDzL8VqXwSfL156jH95620rDWbalb1IxpLCaFksHvtdJSzw3CpuFWaSkoqd1TLKGb26xvFxxz8ACeCpt02vbNbBAaqiuNZqKsYMxUsNO6Jhd0bzngAD29hVrl/yZrX+SVP9ty4SKstIsqFShGrOOZZfX+h0UsFh19q24631TW327ub8oqHDdjZ5ETBwaxvoA9vE85VdRFoB4REQAREQAW8PBB/ajWem01H3o0RNl+1gb1b5I7FO6N89H6iTuRFgrDxNPmiFbmiPBIP5zXnqEf3nrbaIrT8Qd7Dl/o6Z+85xorWx/0Sp/tuXCaIrXRfCR5vqOjsYREVsOCIiAP/2Q==",
  "bodyEncoding": "base64",
  "headers": Object {
    "content-type": Array [
      Object {
        "key": "Content-Type",
        "value": "image/jpeg",
      },
    ],
  },
  "status": "200",
  "statusDescription": "OK",
}
`;

exports[`originResponse handle convert svg to png as default if only size is specified 1`] = `
Object {
  "body": "iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAYAAABw4pVUAAAACXBIWXMAAAsTAAALEwEAmpwYAAAPeklEQVR42u1dCXBV1RkmoB276FStnVartR0qIEFQkS3BpbZVdKzUqlCHutWqU512Wu2ouFRrbatWa1u1FtnCEqMICgYJi0Belkf2kLDkkX0hIQkJJISEvHvvf/r/557z3nk3L+G9kPCSl3Nnztx733KX/zv/es75/1GjTtPGGJMths4BrO14zLxmh7Eu93a2JPVyqGtNB/rMtLyWBabFAjez22gzDh/bZ1Q1bzP3Hlxl5VX+23KXvQLpB/4EaZ4XILP0ZciueAOKapda5Y3J5qGj+cbxE40G/tdwXItZluGlfbdxDNbm3MaWumLhYGsmvz/euwt33xPPPdp+XhgVVRuBob4g7uMEbaD00HpYtHMs27j7Pn7uQ8Dyeg8dzTMLqt+FLcWPwprsm9nKjOmw1DUJWyyz20S2RGn+z2NZQvrVkLTrBpZcuAB2lf0NKpu3mJ3dhwkICTYUVL/H3tv+Xdix/4+8f/APAV4Vz3iG+uxRBYYCCLUviV63yCaAZWwsvBcW7RgLBxrWsWMnGiAHe/qneT/nREVCw7K0yWxF+jVsVeZMtjozLqS2KnMWW5ExjS1Pu5JzAAH5YdaPWGrJM1B/JAuOHC9jH7ivA7oHcp4luKNK4Yoxzucf9huxugJGDJ7HiOMpeJwjJUj9kWxO9A92Xc9Wu+NJhBEYnKirM+MFgWf62sqMGQHnzhb4/SwfSASQ4B72YdaP+X12lb/KbK4B7BxmKx7fq3Sm6AGFXkDKXvli4vgJVTK1dnjY9n2/50S023QOAhEyFOKH2yT30H0IoE/z7mQVTZuZIspIbK3B3dmiU53hFL3DXUzJXnYmto99ihUMM7/qbSI4FyfhcMCpNCf3JKRfxUXatr2/ZW2dNZY0AlCUkviaLEFROtfwBUP2Ljy/AI/zBRbe1o4D8FnBPdy6IkCo1w4mCH2Dw0Ua7xSJ7tkMjQxFwVu0nyPfRdWFw5kzvoMvUyXA6C5v+pwTY3naFCB9MdgcEQoo1EhMrsyYRhYcWWXMYS7Pd3SwoQ9KL2B8C1+iWnJGcd1yFA9XAJmvkeKKULiFOHfb3sfR9O42FVDuGFagqNaU2J+FD18swUB/AkXURNQVs8BWqjOCyvXTAZLzHvJc7le7Z3Nze1PRr5hhdqmgxA8LUKRFJfbCG4cNUkwV1S7h+oJ6X3AAZgX4D6dDqQf6LD2fCUEhZxM2Fz/CfSWhU1pwd5HDyR0WSnyh5IyyxmRmW1GzegVDOG/c205InzpooKicgM8kvPpJvXIqBwU7ksuzUPXk04OEhIakqJJ6Y6rkb/SAQfgWvYgpAmMK21h4H/M0fMyKaxMYxZbIFB1oUPxgTOeNFHdZ42cs/cCLPArgFFt+UOK5TimuXeYDBbcXhmR4JVgPQbbOtWNR3cb6/Luo9/dQ4CoYm4oeol7nE9LtXXXciyanbeBFln1PT8O6gGDj7pr32TLXFX1FBHiIpbGtQAYgaRc75ERXEO54TIqqnIp/2nrDHR+0p9PLExEqm7eKYOIJApEfZ5W/xsWJk0CnCgaFTChGhk4p7wTyfl3eI6ITXBOUs+g5qGNtKJhHkWKhTyBFiOmYISO2hCKXVtXZ2HMOClFlYQ8nnQF9hTBIflc0pfQAxF3210ED5JO8nxFRxT3t+3V2t7CkXTf2ypVCdPEOhgYKcYm0vOY4Y15DhjsQnCckd1BsihR5b76Gqsw/3/2AjzC0tXR4eJBx5SCKrP31HwaIrNzKt4KKLGejTkbAHe9uklbXTiVwGnlAlPjOmcgdpfSQjW2FFslbGRw8mSNGBCJdU1yXwCi2RSFyGekdHKU+g1tyLs+zCEwS277vCWFE9O0HSW+efCl8TlCCkfFSl0QcEIU75srorcvzHA8/hOKJS1AIAHuAaaIQG4Nr9lJbihxB91yWdkUYTukssshgTfYc1m20Sy5ZKiVFxHSJYl2JwRyLy4COEw0GihsKjfQnZBHUSRtcxzA+rHtKBU8drvTQBjGoZTTi7utSdEUEEIcyPw8fqpkerqT+Iwt7OdjjGeIlTkqkGaJFPnQS6ngKchWF633KnSSE5JKIjb8r4mqOFFfb9v4OUDmCr9eFDMrwamRwJO26AcQkChppfFcVWxHWH9Yrwpb3fpR1E0hbfmUUAyK5pOrwF6YQW/kRDTYKkcX1B1pXm+ihDh3NM6R1FNCbogwMG5B4HnikqUjCc29HmlyoRrsj4oNQMBGdrFKpP5Zw/RE3MjgErTShRyyhR2YNqPkbZLpOjLj4GNFGO8c8qFdg7zhCD5RT8SYBIkbeZkQdV/R0MK9kG/LnAc0NEObvAoduddJwtGqF9RopVh08cZExfbGdsCTkTS8H4R+5ShbyqTwUsrbFVvQ2OaXo45xbgHSnUOxPyghwKDRUrFQ/KA6u8M2bEucX4vm1iPxcbDfj8Xhsox26ZJqcRLKp6CFYnDqOe8HL0fON5kaePUUYyNo61nXQK/TIy4ImZyo0pPkE19s0hJ8wYONUs7jHvC9FF6jzpu7AC2xDVjwWMB8WTJpzuw9/+zyeni/+d5P82uV5Ftbm/pRtKJjP1uffPULaPCBnWHDI2woN5+P5TqThcQcNTWzFSLen8fRrTto7UfoGfrnJOS8ZwTHVSWRCXpIjGI+/n+2bigsmUBR1hDUkAchw/Ju4+yruU0OkYS3ufhgAijK4chF+WCmjtciGRmHNIti65zFILvwlDfZDZulfaE4sXbhbXJCOl6uTl9nI3GQoPhlpuFfSsK2zxiio/i9s2fMbQcOH+KRvmkAuaSi2O3uAokxiO0ER0EQxwE/OD1kTy1BeUvwGlTbsLHmKeY0OU5l2yQHBm7Md+5/kUVRU8NHdPAtZqth3dh92dkZjT10CKv84QcPJATSkga40z/M0m0Wd93WVKrJelLNDaChzcep4e96UO943C0SZ9IyKewJsLLyXLgg2O9pWH81E/N+O7zOytNRlAdHZ/BMk2jqrQaw54fTIq/oPe3/nOHtGZgAN4xQajoeU4l/71qnQxHMJxrko+3lgsP5ItkVoBps3pQbgEt3XctDcZTxiAuJBWErRw0AWCHFXqEsGhm+L9+3bu2old0BtiwvESGcINBwHNMztn0sMc0luzZNstnXP41xEhePYHe2s8vEcTSoj7lg1oMOuQznAaNMIAXHQYAoLbSxoOhAwfivNSkRAzPfkOEai+7qQxjF8I2fole8/mDTiAUGRxd//6PHKkAfWJA3tcZX1lhB5+ygw6OKBwbZ8M5RhV0dwjWWVv64BEYDUtWbwcEqYNATUOXKgq4EAybX1R5YVHiB81Iy5S1/RgAhAag7v4N57uDSk5XvCaWwmQHL8gFzFNCCnAsjOfgAykYKyGpChDEiuBkQDogHRgGhANCAaEA2IBkQDogHRgGhANCAaEA2IBkQDogHRgGhANCAaEA2IBkQDogHRgGhANCAaEA2IBkQDogHR87I0IBqQKAPE6kVk6bm9Q2Fub7qdFiPc2e9xYvb7axoQ3+z39P7Nfvel57Bnvy+mE1rkmeieDeGsbSBA9h5crQERgFCBmHD+K9eHeBrWyfUhJbSCaoFcQbW5+BG+QDHUnOy0uIceYsSvoOr0r6Cisk3LQ15BxatCQHtXrVznvoaXkZC5Smpb0yyRuLLX9XHU7PVxE6gQl73GEOw1hrR0eoQuafOtMaxs3goi3wv0lShNrtNUqzAgc9wtszj8Xa7CRXnGV5BSele7lERgfg+Z5ZmSVnYb7XzVqUgqDCMdEFFZzsoqfxVpeJlNw8wgNESwqEMnFy7wLY0GsIrUfCdUxaDEvzR6MRdHpHCoxgflwLJzs/PqaJRMgHV5W5R16hboRZ8BWRqM/Op3eLLM4DScBJRGt9to61F1YZRMJoMfjMXjJn8FnDKTagJ+vvt+qpYGGwrmQ2rJ01B9eLul5D+nLUlkdBjJIkumZ0rBViFp2HKsxEQrFDYW3ido+AtwlTwLNS0uy5HJ4YGATA5KebhL8DjTmTaC6mc4c3XgOSVVuRV/f7U/G9AIBaSzVmYD+rPIF1PYBw1BoSFVhJsrGYNnCeolG9CD5MFT5poeFTLBrBFJVi4W/7tRA+ID5F+Slnj6KJ7nB6lYSr+rpOKVePhN5fdB82U5c2Fdhu02RPJ+bPMoNxa2rzh+c8MIBwQUQN50dm6RY+x2QcO78XgqtrOcmZgC8sYHSe3XZ+5ZkQ3tDHFDmZ7JSil+eMQBQlZTe1edBOR1R0a5UWFnlAsh5+IYZ87FwIzWMEmIRaAUfzIB/0gAhIrArMmeAye8R2WKv6ecRV5INwSjobNM1EAkzJRJMM9Bx7A+ICupO7qTYErOoITRjuzWtzjF0OnO2ysTKX9kJ1JuNURRFBjswl4RTxPrtuNQVc3bZOi8KaL531WxhXtpaZmehrXcC+0t/BINVpUMHW3f9wfmz64Hbw2FVOOqLlkrPX2qAULhlYT0qeDPK9Uz31R/APNXUog/SenuuH5ee2afObIoIEgxKHSYKaueWkrvgohXR3AUkTxPLa9a1pgMVG1NhlucGdkoAGcP5viDlqESzF9rJHi2N7smyOR+XntqH9e29WN2xRtUEUiNVtw6JMoeBSkIdimCUi7NYK953FvZvMXMrvgHt8Ao9CIb5RzcVPQgBCvC1bdlM5WnZqX/q9dTm8vzHKPEk+HUMJFW07rc2/n/A6/5DE8Aur8+yRKhcjVX4j2OaEdkAQkCyjn4cCtCyORJ6fBCqvsUMMKWejkvcXeSDKhANdpX8AKVM8MaTs048FJI2VVRTFHS0GlOTzvildpUO9rhdV6DD/0umsR7sXWIBMK8IbvzF65o2gR2uaG4sIgm6j+R2ACR3Fm5thfEkDRQjw8PkEkszfOcuLbXee0T+B7V6Gt8gu98V5+e9hCrEh2jhmBE0OzbIgwzAdtYqrVBL13ZtNkMFxCS7/lV78gx6Ga83nRsP8AWi+1iJNxGuzBZgRk+ILEIyPMgnDwv+RXYvsfz2wNc4gwbqQUlh2T5VYcHOro3B8myDJfNIf0F5G0JSD3e58sB1wZjlQDEOEVAuvHalwaRCKNV03ZYlPAOAo4s2SAKwBgZAwRIA177fDUBPn72wQACMkGJTcUECS2NGrab+gJItPQBBORcVZYPMCDjg0XBo2LTgGhANCAaEA2IBkQDogHRgAx9QN45rYA4C6FFLSCVgwNIouaQfnNICgEC4QUXwxJZEH5wcWQBEiOItoNeuvrwF6Zd0SeM8LtrIhTWLJKA1NIkCwcgCfRdc/seY0XGtDABiYWM0pfkxOlOCoSqg3CnY/s/mtM3HvCDi0sAAAAASUVORK5CYII=",
  "bodyEncoding": "base64",
  "headers": Object {
    "content-type": Array [
      Object {
        "key": "Content-Type",
        "value": "image/png",
      },
    ],
  },
  "status": "200",
  "statusDescription": "OK",
}
`;

exports[`originResponse handle converts svg to png 1`] = `
Object {
  "body": "iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAYAAABw4pVUAAAACXBIWXMAAAsTAAALEwEAmpwYAAAPeklEQVR42u1dCXBV1RkmoB276FStnVartR0qIEFQkS3BpbZVdKzUqlCHutWqU512Wu2ouFRrbatWa1u1FtnCEqMICgYJi0Belkf2kLDkkX0hIQkJJISEvHvvf/r/557z3nk3L+G9kPCSl3Nnztx733KX/zv/es75/1GjTtPGGJMths4BrO14zLxmh7Eu93a2JPVyqGtNB/rMtLyWBabFAjez22gzDh/bZ1Q1bzP3Hlxl5VX+23KXvQLpB/4EaZ4XILP0ZciueAOKapda5Y3J5qGj+cbxE40G/tdwXItZluGlfbdxDNbm3MaWumLhYGsmvz/euwt33xPPPdp+XhgVVRuBob4g7uMEbaD00HpYtHMs27j7Pn7uQ8Dyeg8dzTMLqt+FLcWPwprsm9nKjOmw1DUJWyyz20S2RGn+z2NZQvrVkLTrBpZcuAB2lf0NKpu3mJ3dhwkICTYUVL/H3tv+Xdix/4+8f/APAV4Vz3iG+uxRBYYCCLUviV63yCaAZWwsvBcW7RgLBxrWsWMnGiAHe/qneT/nREVCw7K0yWxF+jVsVeZMtjozLqS2KnMWW5ExjS1Pu5JzAAH5YdaPWGrJM1B/JAuOHC9jH7ivA7oHcp4luKNK4Yoxzucf9huxugJGDJ7HiOMpeJwjJUj9kWxO9A92Xc9Wu+NJhBEYnKirM+MFgWf62sqMGQHnzhb4/SwfSASQ4B72YdaP+X12lb/KbK4B7BxmKx7fq3Sm6AGFXkDKXvli4vgJVTK1dnjY9n2/50S023QOAhEyFOKH2yT30H0IoE/z7mQVTZuZIspIbK3B3dmiU53hFL3DXUzJXnYmto99ihUMM7/qbSI4FyfhcMCpNCf3JKRfxUXatr2/ZW2dNZY0AlCUkviaLEFROtfwBUP2Ljy/AI/zBRbe1o4D8FnBPdy6IkCo1w4mCH2Dw0Ua7xSJ7tkMjQxFwVu0nyPfRdWFw5kzvoMvUyXA6C5v+pwTY3naFCB9MdgcEQoo1EhMrsyYRhYcWWXMYS7Pd3SwoQ9KL2B8C1+iWnJGcd1yFA9XAJmvkeKKULiFOHfb3sfR9O42FVDuGFagqNaU2J+FD18swUB/AkXURNQVs8BWqjOCyvXTAZLzHvJc7le7Z3Nze1PRr5hhdqmgxA8LUKRFJfbCG4cNUkwV1S7h+oJ6X3AAZgX4D6dDqQf6LD2fCUEhZxM2Fz/CfSWhU1pwd5HDyR0WSnyh5IyyxmRmW1GzegVDOG/c205InzpooKicgM8kvPpJvXIqBwU7ksuzUPXk04OEhIakqJJ6Y6rkb/SAQfgWvYgpAmMK21h4H/M0fMyKaxMYxZbIFB1oUPxgTOeNFHdZ42cs/cCLPArgFFt+UOK5TimuXeYDBbcXhmR4JVgPQbbOtWNR3cb6/Luo9/dQ4CoYm4oeol7nE9LtXXXciyanbeBFln1PT8O6gGDj7pr32TLXFX1FBHiIpbGtQAYgaRc75ERXEO54TIqqnIp/2nrDHR+0p9PLExEqm7eKYOIJApEfZ5W/xsWJk0CnCgaFTChGhk4p7wTyfl3eI6ITXBOUs+g5qGNtKJhHkWKhTyBFiOmYISO2hCKXVtXZ2HMOClFlYQ8nnQF9hTBIflc0pfQAxF3210ED5JO8nxFRxT3t+3V2t7CkXTf2ypVCdPEOhgYKcYm0vOY4Y15DhjsQnCckd1BsihR5b76Gqsw/3/2AjzC0tXR4eJBx5SCKrP31HwaIrNzKt4KKLGejTkbAHe9uklbXTiVwGnlAlPjOmcgdpfSQjW2FFslbGRw8mSNGBCJdU1yXwCi2RSFyGekdHKU+g1tyLs+zCEwS277vCWFE9O0HSW+efCl8TlCCkfFSl0QcEIU75srorcvzHA8/hOKJS1AIAHuAaaIQG4Nr9lJbihxB91yWdkUYTukssshgTfYc1m20Sy5ZKiVFxHSJYl2JwRyLy4COEw0GihsKjfQnZBHUSRtcxzA+rHtKBU8drvTQBjGoZTTi7utSdEUEEIcyPw8fqpkerqT+Iwt7OdjjGeIlTkqkGaJFPnQS6ngKchWF633KnSSE5JKIjb8r4mqOFFfb9v4OUDmCr9eFDMrwamRwJO26AcQkChppfFcVWxHWH9Yrwpb3fpR1E0hbfmUUAyK5pOrwF6YQW/kRDTYKkcX1B1pXm+ihDh3NM6R1FNCbogwMG5B4HnikqUjCc29HmlyoRrsj4oNQMBGdrFKpP5Zw/RE3MjgErTShRyyhR2YNqPkbZLpOjLj4GNFGO8c8qFdg7zhCD5RT8SYBIkbeZkQdV/R0MK9kG/LnAc0NEObvAoduddJwtGqF9RopVh08cZExfbGdsCTkTS8H4R+5ShbyqTwUsrbFVvQ2OaXo45xbgHSnUOxPyghwKDRUrFQ/KA6u8M2bEucX4vm1iPxcbDfj8Xhsox26ZJqcRLKp6CFYnDqOe8HL0fON5kaePUUYyNo61nXQK/TIy4ImZyo0pPkE19s0hJ8wYONUs7jHvC9FF6jzpu7AC2xDVjwWMB8WTJpzuw9/+zyeni/+d5P82uV5Ftbm/pRtKJjP1uffPULaPCBnWHDI2woN5+P5TqThcQcNTWzFSLen8fRrTto7UfoGfrnJOS8ZwTHVSWRCXpIjGI+/n+2bigsmUBR1hDUkAchw/Ju4+yruU0OkYS3ufhgAijK4chF+WCmjtciGRmHNIti65zFILvwlDfZDZulfaE4sXbhbXJCOl6uTl9nI3GQoPhlpuFfSsK2zxiio/i9s2fMbQcOH+KRvmkAuaSi2O3uAokxiO0ER0EQxwE/OD1kTy1BeUvwGlTbsLHmKeY0OU5l2yQHBm7Md+5/kUVRU8NHdPAtZqth3dh92dkZjT10CKv84QcPJATSkga40z/M0m0Wd93WVKrJelLNDaChzcep4e96UO943C0SZ9IyKewJsLLyXLgg2O9pWH81E/N+O7zOytNRlAdHZ/BMk2jqrQaw54fTIq/oPe3/nOHtGZgAN4xQajoeU4l/71qnQxHMJxrko+3lgsP5ItkVoBps3pQbgEt3XctDcZTxiAuJBWErRw0AWCHFXqEsGhm+L9+3bu2old0BtiwvESGcINBwHNMztn0sMc0luzZNstnXP41xEhePYHe2s8vEcTSoj7lg1oMOuQznAaNMIAXHQYAoLbSxoOhAwfivNSkRAzPfkOEai+7qQxjF8I2fole8/mDTiAUGRxd//6PHKkAfWJA3tcZX1lhB5+ygw6OKBwbZ8M5RhV0dwjWWVv64BEYDUtWbwcEqYNATUOXKgq4EAybX1R5YVHiB81Iy5S1/RgAhAag7v4N57uDSk5XvCaWwmQHL8gFzFNCCnAsjOfgAykYKyGpChDEiuBkQDogHRgGhANCAaEA2IBkQDogHRgGhANCAaEA2IBkQDogHRgGhANCAaEA2IBkQDogHRgGhANCAaEA2IBkQDogHR87I0IBqQKAPE6kVk6bm9Q2Fub7qdFiPc2e9xYvb7axoQ3+z39P7Nfvel57Bnvy+mE1rkmeieDeGsbSBA9h5crQERgFCBmHD+K9eHeBrWyfUhJbSCaoFcQbW5+BG+QDHUnOy0uIceYsSvoOr0r6Cisk3LQ15BxatCQHtXrVznvoaXkZC5Smpb0yyRuLLX9XHU7PVxE6gQl73GEOw1hrR0eoQuafOtMaxs3goi3wv0lShNrtNUqzAgc9wtszj8Xa7CRXnGV5BSele7lERgfg+Z5ZmSVnYb7XzVqUgqDCMdEFFZzsoqfxVpeJlNw8wgNESwqEMnFy7wLY0GsIrUfCdUxaDEvzR6MRdHpHCoxgflwLJzs/PqaJRMgHV5W5R16hboRZ8BWRqM/Op3eLLM4DScBJRGt9to61F1YZRMJoMfjMXjJn8FnDKTagJ+vvt+qpYGGwrmQ2rJ01B9eLul5D+nLUlkdBjJIkumZ0rBViFp2HKsxEQrFDYW3ido+AtwlTwLNS0uy5HJ4YGATA5KebhL8DjTmTaC6mc4c3XgOSVVuRV/f7U/G9AIBaSzVmYD+rPIF1PYBw1BoSFVhJsrGYNnCeolG9CD5MFT5poeFTLBrBFJVi4W/7tRA+ID5F+Slnj6KJ7nB6lYSr+rpOKVePhN5fdB82U5c2Fdhu02RPJ+bPMoNxa2rzh+c8MIBwQUQN50dm6RY+x2QcO78XgqtrOcmZgC8sYHSe3XZ+5ZkQ3tDHFDmZ7JSil+eMQBQlZTe1edBOR1R0a5UWFnlAsh5+IYZ87FwIzWMEmIRaAUfzIB/0gAhIrArMmeAye8R2WKv6ecRV5INwSjobNM1EAkzJRJMM9Bx7A+ICupO7qTYErOoITRjuzWtzjF0OnO2ysTKX9kJ1JuNURRFBjswl4RTxPrtuNQVc3bZOi8KaL531WxhXtpaZmehrXcC+0t/BINVpUMHW3f9wfmz64Hbw2FVOOqLlkrPX2qAULhlYT0qeDPK9Uz31R/APNXUog/SenuuH5ee2afObIoIEgxKHSYKaueWkrvgohXR3AUkTxPLa9a1pgMVG1NhlucGdkoAGcP5viDlqESzF9rJHi2N7smyOR+XntqH9e29WN2xRtUEUiNVtw6JMoeBSkIdimCUi7NYK953FvZvMXMrvgHt8Ao9CIb5RzcVPQgBCvC1bdlM5WnZqX/q9dTm8vzHKPEk+HUMJFW07rc2/n/A6/5DE8Aur8+yRKhcjVX4j2OaEdkAQkCyjn4cCtCyORJ6fBCqvsUMMKWejkvcXeSDKhANdpX8AKVM8MaTs048FJI2VVRTFHS0GlOTzvildpUO9rhdV6DD/0umsR7sXWIBMK8IbvzF65o2gR2uaG4sIgm6j+R2ACR3Fm5thfEkDRQjw8PkEkszfOcuLbXee0T+B7V6Gt8gu98V5+e9hCrEh2jhmBE0OzbIgwzAdtYqrVBL13ZtNkMFxCS7/lV78gx6Ga83nRsP8AWi+1iJNxGuzBZgRk+ILEIyPMgnDwv+RXYvsfz2wNc4gwbqQUlh2T5VYcHOro3B8myDJfNIf0F5G0JSD3e58sB1wZjlQDEOEVAuvHalwaRCKNV03ZYlPAOAo4s2SAKwBgZAwRIA177fDUBPn72wQACMkGJTcUECS2NGrab+gJItPQBBORcVZYPMCDjg0XBo2LTgGhANCAaEA2IBkQDogHRgAx9QN45rYA4C6FFLSCVgwNIouaQfnNICgEC4QUXwxJZEH5wcWQBEiOItoNeuvrwF6Zd0SeM8LtrIhTWLJKA1NIkCwcgCfRdc/seY0XGtDABiYWM0pfkxOlOCoSqg3CnY/s/mtM3HvCDi0sAAAAASUVORK5CYII=",
  "bodyEncoding": "base64",
  "headers": Object {
    "content-type": Array [
      Object {
        "key": "Content-Type",
        "value": "image/png",
      },
    ],
  },
  "status": "200",
  "statusDescription": "OK",
}
`;
